packages/cli-platform-ios/src/commands/logIOS/index.ts

```diff
@@ -6,44 +6,42 @@
  *
  */
 
-import {execFileSync, spawnSync} from 'child_process';
+import {spawnSync} from 'child_process';
 import os from 'os';
 import path from 'path';
 import {logger} from '@react-native-community/cli-tools';
-import {Device} from '../../types';
-
-function findAvailableDevice(devices: {[index: string]: Array<Device>}) {
-  for (const key of Object.keys(devices)) {
-    for (const device of devices[key]) {
-      if (device.availability === '(available)' && device.state === 'Booted') {
-        return device;
-      }
-    }
-  }
-  return null;
-}
+import listIOSDevices from '../../tools/listIOSDevices';
+import {getSimulators} from '../runIOS';
 
 /**
  * Starts iOS device syslog tail
  */
 async function logIOS() {
-  const rawDevices = execFileSync(
-    'xcrun',
-    ['simctl', 'list', 'devices', '--json'],
-    {encoding: 'utf8'},
+  // Here we're using two command because first command `xcrun simctl list --json devices` outputs `state` but doesn't return `available`. But second command `xcrun xcdevice list` outputs `available` but doesn't output `state`. So we need to connect outputs of both commands.
+  const simulators = getSimulators();
+  const bootedSimulators = Object.keys(simulators.devices)
+    .map((key) => simulators.devices[key])
+    .reduce((acc, val) => acc.concat(val), [])
+    .filter(({state}) => state === 'Booted');
+
+  const devices = await listIOSDevices();
+  const availableSimulators = devices.filter(
+    ({type, isAvailable}) => type === 'simulator' && isAvailable,
   );
 
-  const {devices} = JSON.parse(rawDevices) as {
-    devices: {[index: string]: Array<Device>};
-  };
+  const bootedAndAvailableSimulators = bootedSimulators.map((booted) => {
+    const available = availableSimulators.find(
+      ({udid}) => udid === booted.udid,
+    );
+    return {...available, ...booted};
+  });
 
-  const device = findAvailableDevice(devices);
-  if (device === null) {
+  if (bootedAndAvailableSimulators.length === 0) {
     logger.error('No active iOS device found');
     return;
   }
 
-  tailDeviceLogs(device.udid);
+  tailDeviceLogs(bootedAndAvailableSimulators[0].udid);
```

Comment:
💡 I think we can implement here prompt if the list of booted and available simulators is greater than 1. WDYT?

Reply:
Whenever we add prompts we need to think of a way for the command to be usable on CI as well, which won't answer the prompt. I'm ok with having a prompt and fall back to `[0]` if in non-interactive environments

Reply:
Also, let's add that separately

Reply:
Created `--interactive` flag to select simulators, by default it selects first one.